src/main/java/de/unijena/cheminf/mortar/controller/HistogramViewController.java

```diff
@@ -59,458 +67,944 @@
+     */
+    private final SimpleBooleanProperty displayBarLabelsSetting;
+    /**
+     * Setting for whether to display bar shadows.
+     */
+    private final SimpleBooleanProperty displayBarShadowsSetting;
+    /**
+     * Setting for whether to display vertical and horizontal grid lines in the histogram.
+     */
+    private final SimpleBooleanProperty displayGridLinesSetting;
+    /**
+     * Setting for whether to display SMILES codes as labels on the y-axis.
+     */
+    private final SimpleBooleanProperty displaySMILESSetting;
+    /**
+     * All settings of this view tool, encapsulated in JavaFX properties for binding in GUI and persistence.
+     */
+    private final List<Property> settings;
     //</editor-fold>
     //
     //<editor-fold desc="private class variables" defaultstate="collapsed">
     /**
-     * HistogramView
+     * Stage of the main application view
+     */
+    private Stage mainStage;
+    /**
+     * HistogramView to display
      */
     private HistogramView histogramView;
     /**
      * Stage of the HistogramView
      */
     private Stage histogramStage;
     /**
-     * Copy of list
+     * Scene of the histogram stage
      */
-    private List<MoleculeDataModel> copyList;
+    private Scene histogramScene;
     /**
-     * Width of the images
+     * List of fragments to display.
+     */
+    private List<FragmentDataModel> fragmentListCopy;
+    /**
+     * Width of molecule depictions displayed when the cursor hovers over a bar. Changes when histogram is resized.
      */
     private double imageWidth;
     /**
-     * Height of the images
+     * Height of molecule depictions displayed when the cursor hovers over a bar. Changes when histogram is resized.
      */
     private double imageHeight;
     /**
-     * Zoom factor of the images
+     * Zoom factor of molecule depictions displayed when the cursor hovers over a bar. Changes when histogram is resized.
      */
     private double imageZoomFactor;
     /**
      * Y-axis of the histogram
      */
     private CategoryAxis categoryAxis;
+    /**
+     * X-axis of the histogram
+     */
+    private NumberAxis numberAxis;
     /**
      * Histogram chart
      */
-   private BarChart histogramChart;
+    private BarChart histogramChart;
     /**
-     * AtomContainer to depict structures
+     * Atom container to depict the respective structure when the cursor hovers over a bar
      */
-   private IAtomContainer atomContainer;
+    private IAtomContainer atomContainerForDisplayCache;
+    //</editor-fold>
+    //
+    //<editor-fold desc="Constructors" defaultstate="collapsed">
     /**
-     * Current value for the number of fragments displayed
+     * Constructor, initialises all settings with their default values. Does *not* open the view.
      */
-    private int displayedFragmentsNumber;
+    public HistogramViewController()  {
+        this.settings = new ArrayList<>(8);
```

Comment:
Thanks for the note on the ArrayList load factor. I have added "+ 1" on some initial capacities now to avoid resizing when I fill the list exactly up to the initial capacity. There are other places in MORTAR where that could be done but which I don't want to touch right now (e.g. FragmentationService, because of Max). We should keep this in mind.

On list vs. array: Well, that was maybe a bad design choice in the beginning. All these setting collections are implemented as lists now (Fragmenters, Settings, now the ViewTool settings). We could maybe change that at a point where we have no open branches.